modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ColorPicker/ColorPicker.es.js

```diff
@@ -86,23 +86,76 @@ const ColorPicker = ({
 	spritemap,
 	value,
 	...otherProps
-}) => (
-	<FieldBase
-		name={name}
-		readOnly={readOnly}
-		spritemap={spritemap}
-		{...otherProps}
-	>
-		<ClayColorPickerWithState
-			inputValue={value ? value : predefinedValue}
+}) => {
+	let _event;
+
+	let previousShow = false;
+
+	const observer = new MutationObserver((mutationsList, observer) => {
+		for (const mutation of mutationsList) {
+			if (
+				mutation.type === 'attributes' &&
+				mutation.attributeName === 'class'
+			) {
+				const show = mutation.target.classList.contains('show');
+
+				if (show === previousShow) {
+					return;
+				}
+
+				if (show) {
+					onFocus(_event);
+				}
+				else {
+					onBlur(_event);
+
+					observer.disconnect();
+				}
+
+				previousShow = show;
+			}
+		}
+	});
+
+	// watch dropdown click for sending to Analytics
+
+	const observeColorDropDown = (event) => {
```

Comment:
@marcosapmf, can we rename it to `handleColorDropDownClicked` instead?

The goal behind this renaming is keep a consistency with the existing code. :wink:

Reply:
@natocesarrego, totally agreed :smile: